### PR TITLE
fix: added reputation sort, permission name

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15618227

image

### CODE REVIEW

File: src/components/frame/v5/pages/VerifiedPage/partials/Table/Table.tsx [hunk 1]
```diff
@@ -13,7 +12,11 @@ import { useVerifiedTable } from './hooks';
 
 const displayName = 'v5.pages.VerifiedPage.partials.Table';
 
-const Table: FC<TableProps> = ({ list, onReputationSortClick }) => {
+const Table: FC<TableProps> = ({
+  list,
+  onReputationSortClick,
+  isSortedDesc,
+}) => {
   // @TODO: Add action for adding new member, removing user from whitelist, handle pagination and sorting
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/Table/Table.tsx:20]: what about this comment?

File: src/components/frame/v5/pages/VerifiedPage/partials/TableItem/TableItem.tsx [hunk 2]
```diff
@@ -26,12 +33,19 @@ const TableItem: FC<TableItemProps> = ({ member, onDeleteClick, onChange }) => {
...
   );
 
-  const { isVerified } = member ?? {};
+  const { isVerified, type } = member ?? {};
 
   const domains = useContributorBreakdown(member);
+  const allRoles = getAllUserRoles(colony, contributorAddress);
+  const permissionRole = getRole(allRoles);
+
+  const userStatus = (type?.toLowerCase() ?? null) as ContributorTypeFilter;
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/TableItem/TableItem.tsx:42]: please avoid casting type

File: src/components/frame/v5/pages/VerifiedPage/partials/TableItem/TableItem.tsx [hunk 3]
```diff
@@ -54,7 +68,9 @@ const TableItem: FC<TableItemProps> = ({ member, onDeleteClick, onChange }) => {
           <Icon name="verified" appearance={{ size: 'tiny' }} />
         </span>
       </div>
-      <div className="hidden sm:flex items-center">status</div>
+      <div className="hidden sm:flex items-center">
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/TableItem/TableItem.tsx:71]: I guess we shoudn't show this <div> when `userStatus`  is undefined

File: src/components/frame/v5/pages/VerifiedPage/partials/TableHead/TableHead.tsx [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
 
+import clsx from 'clsx';
 import Icon from '~shared/Icon';
-import styles from '../Table/Table.module.css';
 import { TableHeadProps } from './types';
 
 const displayName = 'v5.pages.VerifiedPage.partials.TableHead';
 
-const TableHead: FC<TableHeadProps> = ({ onClick }) => {
+const TableHead: FC<TableHeadProps> = ({ onClick, isSortedDesc }) => {
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/TableHead/TableHead.tsx:10]: won't there be an ascending sort option?

File: src/components/frame/v5/pages/VerifiedPage/partials/TableItem/TableItem.tsx (hunk 3 above)
Comment (CzarekDryl) [reply src/components/frame/v5/pages/VerifiedPage/partials/TableItem/TableItem.tsx:71]: I have to show this div empty because some users might not have status and table will break

image

File: src/components/frame/v5/pages/VerifiedPage/partials/Table/Table.tsx (hunk 1 above)
Comment (CzarekDryl) [reply src/components/frame/v5/pages/VerifiedPage/partials/Table/Table.tsx:20]: There are no designs for adding/removing members

File: src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx [hunk 5]
```diff
@@ -19,7 +24,11 @@ const VerifiedPage: FC = () => {
       <TwoColumns aside={<Navigation pageName="members" />}>
         <div className="flex justify-between mt-6">
           <div className="w-full">
-            <Table list={verifiedMembers} onReputationSortClick={() => {}} />
+            <Table
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx:27]: Maybe you should use here the same table which we used in actions? 
`import Table from '~v5/common/Table'; `

File: src/context/MemberContext.tsx [hunk 6]
```diff
@@ -48,10 +50,17 @@ const MemberContextProvider: FC<PropsWithChildren> = ({ children }) => {
   const { colony } = useColonyContext();
   const { domains, colonyAddress = '' } = colony ?? {};
   const isMobile = useMobile();
+  const { pathname } = useLocation();
 
-  const pageSize = isMobile
-    ? HOMEPAGE_MOBILE_MEMBERS_LIST_LIMIT
-    : HOMEPAGE_MEMBERS_LIST_LIMIT;
+  const isVerifiedPage = pathname.includes('verified');
```
Comment (TomaszChmielPagepro) [inline src/context/MemberContext.tsx:55]: Use https://reactrouter.com/en/main/hooks/use-match instead

File: src/context/MemberContext.tsx [hunk 7]
```diff
@@ -48,10 +50,17 @@ const MemberContextProvider: FC<PropsWithChildren> = ({ children }) => {
...
+
+  let pageSize = HOMEPAGE_MEMBERS_LIST_LIMIT;
+
+  if (isVerifiedPage) {
+    pageSize = VERIFIED_MEMBERS_LIST_LIMIT;
+  } else if (isMobile) {
+    pageSize = HOMEPAGE_MOBILE_MEMBERS_LIST_LIMIT;
+  }
```
Comment (TomaszChmielPagepro) [inline src/context/MemberContext.tsx:63]: Wrap with useMemo

File: src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx (hunk 5 above)
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/VerifiedPage.tsx:27]: please use TableBase component

File: src/components/frame/v5/pages/VerifiedPage/partials/Table/Table.tsx [hunk 8]
```diff
@@ -58,8 +57,8 @@ const Table: FC<TableProps> = ({ list, onReputationSortClick }) => {
           </div>
         </div>
       </div>
-      <TableHead onClick={onReputationSortClick} />
-      <div className={styles.tableBody}>
+      <TableHead onClick={onReputationSortClick} isSorted={isSorted} />
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/Table/Table.tsx:60]: this header component should be removed

File: src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React, { FC } from 'react';
+import { useContributorBreakdown } from '~hooks';
+
+import { splitWalletAddress } from '~utils/splitWalletAddress';
+import UserAvatarPopover from '~v5/shared/UserAvatarPopover';
+import { MemberAvatarProps } from './types';
+
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx:7]: missing displayname

File: src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+const MemberAvatar: FC<MemberAvatarProps> = ({ member }) => {
+  const { user } = member || {};
+  const { walletAddress = '', profile } = user || {};
+  const { bio, displayName } = profile || {};
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx:11]: bio = ''

File: src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  const domains = useContributorBreakdown(member);
+
+  return (
+    <div className="ml-1 flex">
+      <UserAvatarPopover
+        userName={displayName}
+        walletAddress={splitWalletAddress(walletAddress || '')}
+        aboutDescription={bio || ''}
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx:19]: you can remove || ''

File: src/components/frame/v5/pages/VerifiedPage/partials/PermissionRow/PermissionRow.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import React, { FC } from 'react';
+
+import { getRole } from '~constants/permissions';
+import { useColonyContext } from '~hooks';
+import Tooltip from '~shared/Extensions/Tooltip';
+import { getAllUserRoles } from '~transformers';
+import { formatText } from '~utils/intl';
+import PermissionsBadge from '~v5/common/Pills/PermissionsBadge';
+import { PermissionRowProps } from './types';
+
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/PermissionRow/PermissionRow.tsx:11]: missing display name

File: src/components/frame/v5/pages/VerifiedPage/partials/PermissionRow/PermissionRow.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+            {formatText(
+              { id: 'role.description' },
+              { role: permissionRole.name },
+            )}
+            <ul className="list-disc font-medium pl-4 mb-4">
+              {permissionRole.permissions.map((permission) => (
+                <li key={permission}>{ColonyRole[permission]}</li>
+              ))}
+            </ul>
+            <a href="/">{formatText({ id: 'learn.more' })}</a>
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/PermissionRow/PermissionRow.tsx:31]: should this redirect to '/'?

File: src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/VerifiedTable.tsx [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+  const onAddClick = () => {};
+  const columns = useVerifiedTableColumns(name);
+
+  const listLength = list.length;
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [rowSelection, setRowSelection] = useState({});
+
+  return (
+    <>
+      <TableHeader
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/VerifiedTable.tsx:37]: I'm wondering if it would be better to add this as a part of the table component? What do you think guys?

File: src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/hooks.tsx [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+        header: () => (
+          <div className="hidden sm:block">
+            {formatText({ id: 'verifiedPage.table.status' })}
+          </div>
+        ),
+        cell: ({ row }) => {
+          const { type } = row.original;
+
+          const userStatus = (type?.toLowerCase() ??
+            null) as ContributorTypeFilter;
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/hooks.tsx:51]: please avoid using casting type

File: src/components/v5/common/TableHeader/TableHeader.tsx [hunk 16]
```diff
@@ -0,0 +1,30 @@
+import clsx from 'clsx';
+import React, { FC, PropsWithChildren } from 'react';
+
+import { TableHeaderProps } from './types';
+
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/TableHeader/TableHeader.tsx:5]: missing displayname

**Review by KornelKwiatkowski — APPROVED** [review]

File: src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx (hunk 10 above)
Comment (CzarekDryl) [reply src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx]: It wasn't working like this, I left it how it was

File: src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx (hunk 11 above)
Comment (CzarekDryl) [reply src/components/frame/v5/pages/VerifiedPage/partials/MemberAvatar/MemberAvatar.tsx:21]: same here

File: src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/hooks.tsx [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+import { ColonyContributorFragment } from '~gql';
+import MemberAvatar from '../MemberAvatar';
+import Checkbox from '~v5/common/Checkbox';
+import { ContributorTypeFilter } from '~v5/common/TableFiltering/types';
+import UserStatusComponent from '~v5/shared/CardWithBios/partials/UserStatus';
+import PermissionRow from '../PermissionRow';
+import { getEnumValueFromKey } from '~utils/getEnumValueFromKey';
+
+export const useVerifiedTableColumns = (
+  name: string,
```
Comment (TomaszChmielPagepro) [inline src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/hooks.tsx:14]: Why is this needed?

File: src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/hooks.tsx [hunk 18]
```diff
@@ -0,0 +1,106 @@
...
+
+  const columns: ColumnDef<ColonyContributorFragment, string>[] = useMemo(
+    () => [
+      columnHelper.display({
+        id: 'checkbox',
+        size: 20,
+        cell: ({ row }) => (
+          <Checkbox
+            id={`verified-${row.original.contributorAddress}`}
+            name={`verified-${row.original.contributorAddress}`}
```
Comment (TomaszChmielPagepro) [inline src/components/frame/v5/pages/VerifiedPage/partials/VerifiedTable/hooks.tsx:30]: I think you can skip these and add default id inside using `useId` like we do in Switch src/components/v5/common/Fields/Switch/Switch.tsx

**Review by krawczykkatarzyna — APPROVED** [review]

**Review by TomaszChmielPagepro — APPROVED** [review]
